### DISCUSSION TITLE
Changing USB MSC descriptors

### DISCUSSION THREAD

**Discussion by victorallume** [micropython/micropython#12243] [STM32 / Pyboard] [answered]
Under Linux, I have a custom MSC-enabled micropython device plugged in, and when I run `ls /dev/disk/by-id` I get `usb-MicroPy_pyboard_Flash_...`. Since it's a custom board, I'd like to distinguish it from the pyboard, but I can't find anywhere in the codebase where I can change this identifier. I've tried all of the #defines that seem appropriate (`MICROPY_PY_SYS_PLATFORM`, `MICROPY_HW_USB_MSC_INTERFACE_STRING` and a few others), but they make no difference on the MSC side (they do, however, change the VCP descriptors). Is this hard-coded somewhere, or have I not found the right place?

**Comment by jimmo** [comment, marked as answer]
@victorallume this comes from here: https://github.com/micropython/micropython/blob/master/ports/stm32/usbd_msc_interface.c#L102  (which is why it's not easily to grep for).

It would be good to make this configurable via mpconfigboard.h (although awkward that it has to be defined this way as uint8 literals).

**Reply by victorallume** [reply]
I've made a PR for this: https://github.com/micropython/micropython/pull/12245

**Reply by victorallume** [reply]
Tried a rebase but still had issues, so I just nuked everything and started a new branch and PR: https://github.com/micropython/micropython/pull/12251
